bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -65,18 +64,16 @@
    * Constructor for BigtableBufferedMutator.
    *
    * @param adapter Converts HBase objects to Bigtable protos
-   * @param settings For bigtable settings
+   * @param bigtableApi For bigtable settings and data client
    * @param listener Handles exceptions. By default, it just throws the exception.
-   * @param session a {@link BigtableSession} object.
    */
   public BigtableBufferedMutator(
       HBaseRequestAdapter adapter,
-      BigtableHBaseSettings settings,
-      BigtableSession session,
+      BigtableApi bigtableApi,
       BufferedMutator.ExceptionListener listener) {
-    helper = new BigtableBufferedMutatorHelper(adapter, settings, session);
+    helper = new BigtableBufferedMutatorHelper(adapter, bigtableApi);
     this.listener = listener;
-    this.host = settings.getDataHost();
+    this.host = bigtableApi.getBigtableHBaseSettings().getDataHost();
```

Comment:
I'd prefer for the bigtable api not expose it's settings. Please pass the settings in parallel to bigtableApi